### ISSUE TITLE
Multi-GPU tarining with DataParallel must has GPU0

### ISSUE THREAD

**Issue by longcw** [pytorch/pytorch#2612]
The loss will never decrease when I train a model using multi-gpus but without gpu0. This is an example I modified from the [official mnist example](https://github.com/pytorch/examples/tree/master/mnist) : https://gist.github.com/longcw/e5efdb0e5a580a6b75604561b6e4f613

```python
gpus = [1, 2]
model = torch.nn.DataParallel(model, device_ids=gpus)
```

If `gpus = [1, 2]` the result after one epoch is 
```
...
Train Epoch: 1 [58240/60000 (97%)]      Loss: 2.363280
Train Epoch: 1 [58880/60000 (98%)]      Loss: 2.373188
Train Epoch: 1 [59520/60000 (99%)]      Loss: 2.351889

Test set: Average loss: 2.3285, Accuracy: 685/10000 (7%)
```

If `gpus = [0, 2]` the result after one epoch is 
```
...
Train Epoch: 1 [58240/60000 (97%)]	Loss: 0.268342
Train Epoch: 1 [58880/60000 (98%)]	Loss: 0.286652
Train Epoch: 1 [59520/60000 (99%)]	Loss: 0.514245

Test set: Average loss: 0.1972, Accuracy: 9436/10000 (94%)
```

**Comment by allenye0119** [comment]
`device_ids` should always contain `0`. If you want to specify which GPUs to use, you can set the environment variable `CUDA_VISIBLE_DEVICES`, either by running your script like this:
```
CUDA_VISIBLE_DEVICES=1,2 python your_script.py
```
or set it programmatically in your script like this:
```
os.environ['CUDA_VISIBLE_DEVICES'] = '1, 2'
```
So that device_ids 0, 1 will map to GPUs 1, 2, respectively.
(Note that for the second method, you should put that line before any cuda relevant function call.)

**Comment by longcw** [comment]
@allenye0119 Thank you.
